src/main/java/entities/GuestUser.java

```diff
@@ -0,0 +1,8 @@
+package entities;
+
+public class GuestUser extends User{
```

Comment:
This inheritance relationship should be the other way around since guestUser has less functionality than User